### ISSUE TITLE
Better compatibility of Stick add to cart header with complex product types (Mix and Match, etc)

### ISSUE THREAD

**Issue by helgatheviking** [woocommerce/storefront#1308]
## Problem

The sticky "add to cart" banner isn't compatible with complex product types.. or products that need configuration. As far as i can tell for Mix and Match, it shows the shop loop add to cart text and url... which in the case of MNM is "select options" and links to the single product page that you are currently on. And it cannot actually add the product to the cart without the configuration in the form data anyway. 

![image](https://user-images.githubusercontent.com/507025/78800650-ebf59c80-7978-11ea-9336-2fcbfc7ec9b1.png)

## Describe the solution you'd like

The `storefront_sticky_single_add_to_cart()` function seems to check only that the product is purchasable, in stock, or an external type.

```
		$show = false;

		if ( $product->is_purchasable() && $product->is_in_stock() ) {
			$show = true;
		} else if ( $product->is_type( 'external' ) ) {
			$show = true;
		}

		if ( ! $show ) {
			return;
		}
```

While the button itself is not actually using ajax, the `supports('ajax_add_to_cart')` might be a good proxy for whether this product can be added to the cart without configuration via this sticky header button. 

## Describe alternatives you've considered

In the meantime, this snippet removes the hook for Mix and Match product pages, but it would be preferable to not need it and have it happen automatically.

```
/**
 * Remove stick add to cart bar for Mix and Match product types
 */
function kia_remove_sticky_add_to_cart_for_mnm() {

	global $product;

	if( $product instanceof WC_Product && $product->is_type( 'mix-and-match' ) ) {
		remove_action( 'storefront_after_footer', 'storefront_sticky_single_add_to_cart', 999 );
	}
	
}
add_action( 'storefront_after_footer', 'kia_remove_sticky_add_to_cart_for_mnm' );
```

**Comment by haszari** [comment]
Thanks for reporting this @helgatheviking !

Just to clarify, does this issue affect WooCommerce + Storefront or does it require another extension, for example [Mix & Match Products](https://woocommerce.com/products/woocommerce-mix-and-match-products/)?

I tested with a variable product (`Hoodie`) from the sample Woo product data. The sticky header shows `Select options`, and when clicked it scrolls the options and `Add to cart` button into view. That feels like a useful feature on long product page, even if direct add-to-cart is not possible for the product.

**Comment by helgatheviking** [comment]
Yes this is WooCommerce + Storefront + Mix and Match (I'm the author). With a Mix and Match product the sticky header "select options" button doesn't scroll to the add to cart button. it does appear to scroll to the top of the form where you make your selections... I thought it was reloading the page, but I think I misremembered that. 

Here's what it looks like:
https://share.getcloudapp.com/7Ku0gjb6

the form can be quite long itself so the add to cart button may not be in view. In the case where the form is quite long, I'm not sure this is adding a lot of value, but honestly I can't think of where else you'd scroll to, so it's probably ok after all. 

Also re-visting my customer's support ticket... it looks like he was expecting it to add to the product to the cart even though the button does clearly say "select options". So that might've gotten me turned around a bit.
